tests/languages/docker_test.py

```diff
@@ -12,3 +16,75 @@ def invalid_attribute():
         getgid=invalid_attribute,
     ):
         assert docker.get_docker_user() == ()
+
+
+class TestInDocker:
+    @mock.patch('builtins.open', new_callable=mock_open)
+    def test_in_docker_no_file(self, mock_file):
+        mock_file.side_effect = FileNotFoundError
+
+        assert docker.is_in_docker() is False
+        mock_file.assert_called()
+
+    @mock.patch('builtins.open', new_callable=mock_open, read_data='tdockert')
+    def test_in_docker_docker_in_file(self, _):
+        assert docker.is_in_docker() is True
+
+    @mock.patch('builtins.open', new_callable=mock_open, read_data='testtest')
+    def test_in_docker_docker_not_in_file(self, _):
+        assert docker.is_in_docker() is False
+
+
+class TestDockerPath:
+    @mock.patch.object(docker, 'is_in_docker', return_value=False)
+    def test_not_in_docker_returns_same(self, _):
+        assert docker.get_docker_path('abc') == 'abc'
+
+    @mock.patch.object(docker, 'is_in_docker', return_value=True)
+    def test_in_docker_no_binds_same_path(self, _):
+        binds_list: List[str] = []
+        output_string = json.dumps([{'HostConfig': {'Binds': binds_list}}])
+        with mock.patch.object(
+            subprocess, 'check_output',
+            return_value=output_string,
+        ):
+            assert docker.get_docker_path('abc') == 'abc'
+
+    @mock.patch.object(docker, 'is_in_docker', return_value=True)
+    def test_in_docker_binds_path_equal(self, _):
+        binds_list = [
+            '/opt/my_code:/project',
+        ]
+        output_string = json.dumps([{'HostConfig': {'Binds': binds_list}}])
+        with mock.patch.object(
+            subprocess, 'check_output',
+            return_value=output_string,
+        ):
+            assert docker.get_docker_path('/project') == '/opt/my_code'
+
+    @mock.patch.object(docker, 'is_in_docker', return_value=True)
```

Comment:
this should be a fixture that gets reused in each of the tests

Reply:
done